sklearn/cluster/bicluster.py

```diff
@@ -35,10 +35,19 @@ def _scale_normalize(X):
 
     """
     X = make_nonnegative(X)
-    row_diag = np.asarray(1.0 / np.sqrt(X.sum(axis=1))).squeeze()
-    col_diag = np.asarray(1.0 / np.sqrt(X.sum(axis=0))).squeeze()
+    row_diag = np.asarray(1.0 / np.sqrt(X.sum(axis=1)))
+    if row_diag.shape[0] != 1:
```

Comment:
You could run squeeze and then reshape. Or, probably better, make sure ``min_samples=2`` in ``check_array``